packages/contracts-noir/test/Semaphore.ts

```diff
@@ -0,0 +1,672 @@
+            )
+        })
+    })
+
+    describe("SemaphoreGroups", () => {
+        describe("# hasMember", () => {
+            it("Should return true because the member is part of the group", async () => {
+                const { semaphoreContract, accountAddresses, groupId } = await loadFixture(deploySemaphoreFixture)
+
+                const members = Array.from({ length: 3 }, (_, i) => new Identity(i.toString())).map(
+                    ({ commitment }) => commitment
+                )
+
+                await semaphoreContract["createGroup(address)"](accountAddresses[0])
+                await semaphoreContract.addMember(groupId, members[0])
+
+                const isMember = await semaphoreContract.hasMember(groupId, members[0])
+
+                await expect(isMember).to.be.true
+            })
+
+            it("Should return false because the member is not part of the group", async () => {
+                const { semaphoreContract, accountAddresses, groupId } = await loadFixture(deploySemaphoreFixture)
+
+                await semaphoreContract["createGroup(address)"](accountAddresses[0])
+
+                const identity = new Identity()
+                const isMember = await semaphoreContract.hasMember(groupId, identity.commitment)
+
+                await expect(isMember).to.be.false
+            })
+        })
+
+        describe("# indexOf", () => {
+            it("Should return the index of a member", async () => {
+                const { semaphoreContract, accountAddresses, groupId } = await loadFixture(deploySemaphoreFixture)
+
+                const members = Array.from({ length: 3 }, (_, i) => new Identity(i.toString())).map(
+                    ({ commitment }) => commitment
+                )
+
+                await semaphoreContract["createGroup(address)"](accountAddresses[0])
+                await semaphoreContract.addMember(groupId, members[0])
+
+                const index = await semaphoreContract.indexOf(groupId, members[0])
+
+                await expect(index).to.equal(0)
+            })
+        })
+
+        describe("# getMerkleTreeDepth", () => {
+            it("Should return the merkle tree depth", async () => {
+                const { semaphoreContract, accountAddresses, groupId } = await loadFixture(deploySemaphoreFixture)
+
+                const members = Array.from({ length: 3 }, (_, i) => new Identity(i.toString())).map(
+                    ({ commitment }) => commitment
+                )
+
+                await semaphoreContract["createGroup(address)"](accountAddresses[0])
+                await semaphoreContract.addMember(groupId, members[0])
+                await semaphoreContract.addMember(groupId, members[1])
+                await semaphoreContract.addMember(groupId, members[2])
+
+                const depth = await semaphoreContract.getMerkleTreeDepth(groupId)
+
+                await expect(depth).to.equal(2)
+            })
+        })
+    })
+})
```

Comment:
This comment is a bit in line with what I said earlier, about the duplicate contract; is there a better way to structure the re-usage of the `SemaphoreGroups.sol` contract, also so it doesn't clutter up this test file? I'm not sure, but I think it would be good to think about since it would mean less repetitive code